src/platforms/common/blackpill-f4/meson.build

```diff
@@ -61,7 +61,7 @@ if blackpill_alternative_pinout != '0'
 endif
 
 if bmd_bootloader
-	probe_blackpill_args += ['-DBMD_BOOTLOADER']
+	probe_blackpill_args += ['-DBMP_BOOTLOADER']
```

Comment:
As much as this works, we feel that fixing the nomenclature in the code would be better here - after all, only the bootloader for native can really be called the "BMP bootloader".

Reply:
Added the rename in a separate commit.
May I add some potential OTG reset bugfix to this PR later this week, or you'd like to merge this as-is and deal with that later? I'd like to see this working properly (including "OOBE") before v2.0-rc0 hits or something.

Reply:
As this is a self-contained change right now, going to merge as-is and we'll one of us deal with that in its own PR - separation of concerns. It'd be great to see the Black Pill platforms properly fixed for v2.0, so you're welcome to PR away once you know what the fix is (as we appreciate you have Black Pill units and so can actually test the platform).